### DISCUSSION TITLE
Access map reduce figures after old source docs have been deleted

### DISCUSSION THREAD

**Discussion by chris-fellows** [ravendb/ravendb#15154] [Q&A] [answered]
RavenDB 5.4.2 (AWS cluster, Dot NET Core, C#)
For my reporting on a particular collection then I currently get the aggregated data from a map reduce index. The data is
aggregated by hour and doesn't get modified after the creation day. I want to be able to delete documents older than N days but still be able to report on the aggregated data from both current & deleted documents. I've tried outputting the map reduce index to a new collection but if I delete the source docs then the new collection docs are updated.

Is my only way to accomplish what I want to manually periodically export the map reduce docs to a new collection?
When I want to create the report then do I need to concatenate the map reduce docs and the exported map reduce docs in my code or is there a way to create a virtual collection that concatenates both sets of docs together for me?

**Comment by DejanMilicic** [comment, marked as answer]
Can you 
1. establish RavenDB ETL from the database to itself, replicating documents to a new collection
2. turning off deletes in replication https://ravendb.net/docs/article-page/5.4/csharp/server/ongoing-tasks/etl/raven#filtering-out-all-deletions

**Reply by chris-fellows** [reply]
Do you mean to set the map reduce index on the source doc to output to collection A? And then create an ETL script that copies collection A (map reduce output) to collection B and that disables deletes of collection B? If source docs are only updated on creation day then it is OK to keep the DB smaller by deleting docs from collection A where the doc doesn't aggregate any source docs created today?

The source collection has about 1 million docs and has a high volume added each day.

**Reply by DejanMilicic** [reply]
Yes

1. Map/Reduce creates artificial collection A
2. ETL operating from the database into itself, replicating from A to B
3. this replication has a script as referenced above that prevents tombstones to be replicated from A to B when document is deleted
4. when replicating to B, you can introduce @expires on documents replicated to B, so that B will have rolling cleanup over time

**Reply by chris-fellows** [reply]
Can I periodically delete selected docs from collection A (or set the expiry) when I know that there won't be any more updates to the related source docs? Would it cause any performance issues for the server?

**Reply by DejanMilicic** [reply]
Collection A is an artificial one - it will be maintained by map/reduce index which is ordered to materialize into this collection.
So, as you are deleting documents from collection X (assuming that map/reduce index is processing documents from collection X), collection A will be automatically updated.

It is always hard to talk about performance. It is usually defined via SLA, and you also need to take into account other parameters like number of requests per second, number of writes, etc. Best would be to try out some demanding load testing to establish baseline SLA

**Reply by chris-fellows** [reply]
If I disable deletes on collection A (map reduce output) then collection B increases in size whenever changes are made to the source doc. It leaves the old collection B doc and creates a new one with the updated values. This will cause problems for my reporting.

Is there any way to overcome this issue?

My aim is to effectively take a snapshot of section of the map reduce index (usually a date range) so that I can include those statistics to report on old periods where the underlying source docs have expired. I can accomplish it by creating a background task in my code that periodically reads the map reduce index for a date range and saves the results to collection B but I wanted to do the work on the Raven server if possible.
